pkg/kms/barbican/barbican.go

```diff
@@ -0,0 +1,95 @@
+package barbican
+
+import (
+	"github.com/golang/glog"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/keymanager/v1/secrets"
+)
+
+type BarbicanService interface {
+	GetSecret(cfg Config) ([]byte, error)
+}
+
+type KMSOpts struct {
+	KeyID string `gcfg:"key-id"`
+}
+
+//Config to read config options
+type Config struct {
+	Global struct {
+		AuthURL    string `gcfg:"auth-url"`
+		Username   string
+		UserID     string `gcfg:"user-id"`
+		Password   string
+		TenantID   string `gcfg:"tenant-id"`
+		TenantName string `gcfg:"tenant-name"`
+		DomainID   string `gcfg:"domain-id"`
+		DomainName string `gcfg:"domain-name"`
```

Comment:
Can you add support of trustee and application credentials?
Using a trust-id is supported by gophercloud.

Reply:
https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/share/manila/shareoptions/openstack.go#L27

Reply:
@strigazi will add Trust support.

Reply:
Is support for trust-id added in another PR? Did you find any difficulties?